src/reducers/_portis.js

```diff
@@ -0,0 +1,128 @@
+import {apiGetPortisNetwork} from '../helpers/api';
+import {parseError} from '../helpers/utilities';
+import {
+  accountUpdateAccountAddress,
+  accountUpdateNetwork,
+  accountUpdateWeb3,
+} from './_account';
+import {notificationShow} from './_notification';
+import {PortisProvider} from 'portis';
+import Web3 from 'web3';
+// -- Constants ------------------------------------------------------------- //
+const PORTIS_CONNECT_REQUEST = 'portis/PORTIS_CONNECT_REQUEST';
+const PORTIS_CONNECT_SUCCESS = 'portis/PORTIS_CONNECT_SUCCESS';
+const PORTIS_CONNECT_FAILURE = 'portis/PORTIS_CONNECT_FAILURE';
+
+const PORTIS_NOT_AVAILABLE = 'portis/PORTIS_NOT_AVAILABLE';
+
+const PORTIS_UPDATE_PORTIS_ACCOUNT = 'portis/PORTIS_UPDATE_PORTIS_ACCOUNT';
+
+// -- Actions --------------------------------------------------------------- //
+
+let accountInterval = null;
+
+export const updateAccountAddress = accountAddress => dispatch => {
+  if (accountAddress) {
+    console.log('ACCOUNT ADDRESS', accountAddress);
+    dispatch(accountUpdateAccountAddress(accountAddress, 'PORTIS'));
+    window.browserHistory.push('/domains');
+  }
+};
+
+export const portisUpdatePortisAccount = address => (dispatch, getState) => {
+  if (address !== getState().portis.accountAddress) {
+    dispatch({
+      type: PORTIS_UPDATE_PORTIS_ACCOUNT,
+      payload: address,
+    });
+    dispatch(updateAccountAddress(address));
+  }
+};
+
+export const portisConnectInit = () => (dispatch, getState) => {
+  const web3 = new Web3(
+    new PortisProvider({
+      apiKey: '1cd61de82681d63b30620c48339f7c97',
```

Comment:
Before merging this, let's create a new apiKey and replace this with `process.env.REACT_APP_PORTIS_API_KEY`

Reply:
This key is basically already compromised, that's why we should create a new one. Then we can add on Netlify the environment variable `REACT_APP_PORTIS_API_KEY`

Reply:
The Portis API key is not a secret and there wouldn't be a way to hide it since it's on the client.

"The API Key is not secret (as it is being sent from the client), its goal is to verify that the dApp using Portis has been registered and is approved." - https://github.com/portis-project/portis-sdk

Reply:
Cool then!

Reply:
Just going to go ahead and do some UI changes first